assets/js/components/conversion-tracking/ConversionTrackingToggle.js

```diff
@@ -0,0 +1,145 @@
+ */
+import {
+	createInterpolateElement,
+	useState,
+	Fragment,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Switch } from 'googlesitekit-components';
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import ErrorText from '../../components/ErrorText';
+import Link from '../Link';
+import LoadingWrapper from '../LoadingWrapper';
+import ConfirmDisableConversionTrackingDialog from './ConfirmDisableConversionTrackingDialog';
+import { useFeature } from '../../hooks/useFeature';
+import PropTypes from 'prop-types';
+
+const { useDispatch, useSelect } = Data;
+
+export default function ConversionTrackingToggle( { loading } ) {
+	const iceEnabled = useFeature( 'conversionInfra' );
+	const [ saveError ] = useState( null );
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+
+	const isConversionTrackingEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).isConversionTrackingEnabled()
+	);
+
+	const isSaving = useSelect( ( select ) =>
+		select( CORE_SITE ).isFetchingSaveConversionTrackingSettings()
+	);
+
+	const { setConversionTrackingEnabled } = useDispatch( CORE_SITE );
+
+	if ( ! iceEnabled ) {
+		return null;
+	}
+
+	return (
+		<Fragment>
+			<div>
+				{
+					<LoadingWrapper
+						loading={ loading }
+						width="180px"
+						height="21.3px"
+					>
+						<Switch
+							label={ __(
+								'Enable enhanced conversion tracking',
+								'google-site-kit'
+							) }
+							checked={ isConversionTrackingEnabled }
+							disabled={ isSaving || loading }
+							onClick={ () => {
+								// If Conversion Tracking is currently enabled, show a confirmation
+								// dialog warning users about the impact of disabling it.
+								if ( isConversionTrackingEnabled ) {
+									setShowConfirmDialog( true );
+								} else {
+									// Conversion Tracking is not currently enabled, so this toggle
+									// enables it.
+									setConversionTrackingEnabled( true );
+								}
+							} }
+							hideLabel={ false }
+						/>
+					</LoadingWrapper>
+				}
+				{ saveError && <ErrorText message={ saveError.message } /> }
+				{
+					<LoadingWrapper
+						className="googlesitekit-settings-conversion-tracking-switch-description--loading"
+						loading={ loading }
+						width="750px"
+						height="42px"
+						smallWidth="386px"
+						smallHeight="84px"
+						tabletWidth="540px"
+						tabletHeight="84px"
+					>
+						<p className="googlesitekit-settings-module__fields-group-helper-text">
+							{ createInterpolateElement(
+								__(
+									'Conversion tracking allows you to measure additional events on your site from other plugins that Site Kit integrates with to optimize your campaign performance. <a>Learn more</a>',
+									'google-site-kit'
+								),
+								{
+									br: <br />,
+									a: (
+										<Link
+											href="#"
```

Comment:
This link doesn't go anywhere, are we waiting for a support URL? If so let's add a `TODO` comment, file an issue about it, and note it here.

Reply:
Thanks for the reminder on this one, ya we actually do not have a URL defined for this, and it is not really an official _thing_ so I will need clarity on what we need here, as it is in the designs. Will reach out internally, ty.

Reply:
@tofumatt I've added a TODO here. Will try to source a URL today, or we should remove the link entirely.

Reply:
@tofumatt I've now updated this to contain the applicable internal docs link (in progress with the support team), using the slug provided by them, `enhanced-conversion-tracking`.